File: java/src/Problem6.java
import java.util.LinkedList;
import java.util.List;

/**
 * Created by adam on 2/1/14.
 */
public class Problem6 {
    public String answer() {
        int min = 1;
        int max = 100;
        List<Integer> l = new LinkedList<Integer>();
        for(int i = min; i <= max; i++) {
            l.add(i);
        }
        return Integer.toString(sumOfN(l) - sumOfSquares(l));
    }
    private int sumOfSquares(List<Integer> l) {
        int sum = 0;
        for(int i : l) {
            sum += Math.pow(i, 2);
        }
        return sum;
    }
    private int sumOfN(List<Integer> l) {
        int sum = 0;
        for(int i : l) {
            sum += i;
        }
        return (int)Math.pow(sum, 2);
    }

}

File: dart/lib/src/Problem6.dart
import 'Euler.dart';
import 'dart:math';

class Problem6 implements Euler {
  List<int> gen(int n) {
    return new List<int>.generate(n, (i) => i + 1);
  }

  int sumOfSquares(int n) {
    return gen(n).map((i) => pow(i, 2)).fold(0, (i, acc) => i + acc);
  }

  int squareOfSum(int n) {
    return pow(gen(n).fold(0, (i, acc) => i + acc), 2);
  }

  int differenceBetweenThem(int n) {
    return squareOfSum(n) - sumOfSquares(n);
  }

  String answer() {
    return differenceBetweenThem(100).toString();
  }
}
